NineChronicles.Headless/GraphTypes/StateQuery.cs

```diff
@@ -346,7 +349,136 @@ public StateQuery()
+                        ? new RuneSlotState(myRawRuneSlotState)
+                        : new RuneSlotState(BattleType.Arena);
+
+                    var myRuneStates = new List<RuneState>();
+                    var myRuneSlotInfos = myRuneSlotState.GetEquippedRuneSlotInfos();
+                    foreach (var address in myRuneSlotInfos.Select(info => RuneState.DeriveAddress(myAvatarAddress, info.RuneId)))
+                    {
+                        if (context.Source.TryGetState(address, out List rawRuneState))
+                        {
+                            myRuneStates.Add(new RuneState(rawRuneState));
+                        }
+                    }
+
+                    //Enemy
+                    var enemyArenaAvatarStateAdr = ArenaAvatarState.DeriveAddress(enemyAvatarAddress);
+                    var enemyArenaAvatarState = context.Source.GetArenaAvatarState(enemyArenaAvatarStateAdr, enemyAvatar);
+                    List<Guid> enemyArenaEquipementList = enemyAvatar.inventory.Equipments.Where(f=>enemyArenaAvatarState.Equipments.Contains(f.ItemId)).Select(n => n.ItemId).ToList();
+                    List<Guid> enemyArenaCostumeList = enemyAvatar.inventory.Costumes.Where(f=>enemyArenaAvatarState.Costumes.Contains(f.ItemId)).Select(n => n.ItemId).ToList();
+
+                    var enemyRuneSlotStateAddress = RuneSlotState.DeriveAddress(enemyAvatarAddress, BattleType.Arena);
+                    var enemyRuneSlotState = context.Source.TryGetState(enemyRuneSlotStateAddress, out List enemyRawRuneSlotState)
+                        ? new RuneSlotState(enemyRawRuneSlotState)
+                        : new RuneSlotState(BattleType.Arena);
+
+                    var enemyRuneStates = new List<RuneState>();
+                    var enemyRuneSlotInfos = enemyRuneSlotState.GetEquippedRuneSlotInfos();
+                    foreach (var address in enemyRuneSlotInfos.Select(info => RuneState.DeriveAddress(enemyAvatarAddress, info.RuneId)))
+                    {
+                        if (context.Source.TryGetState(address, out List rawRuneState))
+                        {
+                            enemyRuneStates.Add(new RuneState(rawRuneState));
+                        }
+                    }
+
+                    var myArenaPlayerDigest = new ArenaPlayerDigest(
+                        myAvatar,
+                        myArenaEquipementList,
+                        myArenaCostumeList,
+                        myRuneStates);
 
+                    var enemyArenaPlayerDigest = new ArenaPlayerDigest(
+                        enemyAvatar,
+                        enemyArenaEquipementList,
+                        enemyArenaCostumeList,
+                        enemyRuneStates);
+
+                    Random rnd  =new Random();          
+
+                    int win = 0;
+                    int loss = 0;
+
+                    for (var i = 0; i < 1000; i++)
```

Comment:
it would be nice if we have an option for this iteration count as configurable.

Reply:
introducing a hard cap would be nice! I don't have an appropriate number for that, but both 1k and 5k are looks good to me.